tests/Test/Types/AttributeTest.php

```diff
@@ -57,81 +58,80 @@ public function attribute_value_can_be_an_expression()
 EOD
         );
 
-        $this->assertEquals("mysql:host=localhost;dbname=application", run('speak'));
+        $this->assertEquals("mysql:host=localhost;dbname=application", $this->ws('speak')->getOutput());
     }
 
     /** @test */
     public function isset_returns_false_when_attribute_is_not_defined()
     {
-        Fixture::workspace(<<<'EOD'
+        $this->workspace()->put('workspace.yml', <<<'EOD'
 command('isset'): |
   #!php
   echo (isset($ws['message'])) ? 'yes' : 'no';
 EOD
         );
 
-        $this->assertEquals('no', run('isset'));
+        $this->assertEquals('no', $this->ws('isset')->getOutput());
     }
 
     /** @test */
     public function isset_returns_true_when_attribute_is_defined_and_has_a_value()
     {
-        Fixture::workspace(<<<'EOD'
+        $this->workspace()->put('workspace.yml', <<<'EOD'
 attribute('message'): Hello World
 command('isset'): |
   #!php
   echo (isset($ws['message'])) ? 'yes' : 'no';
 EOD
         );
 
-        $this->assertEquals('yes', run('isset'));
+        $this->assertEquals('yes', $this->ws('isset')->getOutput());
     }
 
     /** @test */
     public function isset_returns_true_even_when_attribute_value_is_null()
     {
-        Fixture::workspace(<<<'EOD'
+        $this->workspace()->put('workspace.yml', <<<'EOD'
 attribute('message'): null
 command('isset'): |
   #!php
   echo (isset($ws['message'])) ? 'yes' : 'no';
 EOD
         );
 
-        $this->assertEquals('yes', run('isset'));
+        $this->assertEquals('yes', $this->ws('isset')->getOutput());
     }
 
     /** @test */
     public function null_values_are_also_represented_internally_as_null()
     {
-        Fixture::workspace(<<<'EOD'
+        $this->workspace()->put('workspace.yml', <<<'EOD'
 attribute('message'): null
 command('isnull'): |
   #!php
   echo (is_null($ws['message'])) ? 'yes' : 'no';
 EOD
         );
 
-        $this->assertEquals('yes', run('isnull'));
+        $this->assertEquals('yes', $this->ws('isnull')->getOutput());
     }
 
     /** @test */
     public function attribute_precedence_is_respected()
     {
-        $path = Fixture::sampleData('attribute/precedence');
+        $this->workspace()->loadSample('attribute/precedence');
 
-        chdir($path.'/workspace');
+        // @todo: Make this use a dataprovider
+        $this->assertEquals('Hello From harness.default',  $this->ws('get "key.1"', 'workspace')->getOutput());
+        $this->assertEquals('Hello From harness.normal',   $this->ws('get "key.2"', 'workspace')->getOutput());
+        $this->assertEquals('Hello From harness.override', $this->ws('get "key.3"', 'workspace')->getOutput());
```

Comment:
needed to specify a subpath for each aassertion here whicih is ugly - a data provider would solve this.

Reply:
(i.e. do this before merging)